File: exam_portal/src/main/java/com/exam/exam_portal/services/QuizService.java
package com.exam.exam_portal.services;

import com.exam.exam_portal.models.Exam.Category;
import com.exam.exam_portal.models.Exam.Quiz;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Set;

public interface QuizService {

    public Quiz addQuiz(Quiz quiz);

    public Quiz updateQuiz(Quiz quiz);

    public Set<Quiz> getQuizzes();

    public Quiz getQuiz(Long quizid);

    public void deleteQuiz(Long quizId);


    public List<Quiz> getquizofcategory(Category category);

    public List<Quiz> getActiveQuiz();

    public List<Quiz> getActiveQuizofCategory(Category category);
}

File: client/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http:HttpClient) { }

  public getQuiz(){
    return this.http.get(`${baseUrl}/quiz/`)
  }

  public addquiz(quiz:any){
    return this.http.post(`${baseUrl}/quiz/`, quiz)
  }

  public deletequiz(qid:any){
    return this.http.delete(`${baseUrl}/quiz/${qid}`);
  }

  public getAQuiz(qid:any){
    return this.http.get(`${baseUrl}/quiz/${qid}`);
  }

  public updatequiz(quiz:any){
    return this.http.put(`${baseUrl}/quiz/`,quiz);
  }

  public getQuizofCategory(cid:any){
    return this.http.get(`${baseUrl}/quiz/category/${cid}`)
  }
}
